Tasks/Gradle/CodeAnalysis/Common/BuildOutput.ts

```diff
@@ -0,0 +1,52 @@
+import {ModuleOutput} from './ModuleOutput';
+
+import path = require('path');
+import fs = require('fs');
+import glob = require('glob');
+
+import tl = require('vsts-task-lib/task');
+
+
+
+/**
+ * Build output from a single or multi module project. Identifies modules based on path conventions. 
+ * 
+ * @export
+ * @class BuildOutput
+ * @implements {IBuildOutput}
+ */
+export class BuildOutput {
+
+    private moduleOutputs: ModuleOutput[] = [];
+
+    constructor(private rootDirectory: string, private buildOutputDirName: string) {
```

Comment:
The only difference between Maven in Gradle builds is where the output is generated - for Gradle this is "build" and for "Maven" this is "target" sub directories.